src/backend/api/__init__.py

```diff
@@ -127,30 +129,30 @@ async def conn(self, attempts: int = 1) -> AsyncGenerator[S3Client, None]:
             except Exception as e:
                 logger.error(f"Unknown error: {e}")
                 logger.error(f"  exception: {type(e)}")
-                raise HTTPException(status_code=500)
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
+                )
 
 
 def decode_client_error(e: ClientError) -> Tuple[int, str]:
     """
-    Returns a tuple of `(status_code, error_message)` according to the
+    Returns a tuple of `(status_code, detail)` according to the
     `botocore's ClientError` exception that is passed as an argument.
     """
-    status_code = 500
-    msg = "Unknown Error"
-
-    if "Error" in e.response:
-        if "Code" in e.response["Error"]:
-            if e.response["Error"]["Code"] == "InvalidAccessKeyId":
-                msg = "Invalid credentials"
-                status_code = status.HTTP_401_UNAUTHORIZED
-            elif e.response["Error"]["Code"] == "NoSuchKey":
-                msg = "No such key"
-                status_code = status.HTTP_404_NOT_FOUND
-            elif e.response["Error"]["Code"].isnumeric():
-                status_code = int(e.response["Error"]["Code"])
-                if "Message" in e.response["Error"]:
-                    msg = e.response["Error"]["Message"]
-    return (status_code, msg)
+    status_code = pydash.get(
+        e.response,
+        "ResponseMetadata.HTTPStatusCode",
+        status.HTTP_500_INTERNAL_SERVER_ERROR,
+    )
+    detail = None
+    code = pydash.get(e.response, "Error.Code")
+    if code is not None:
+        if code.isnumeric():
+            status_code = int(code)
+            detail = pydash.get(e.response, "Error.Message")
+        else:
+            detail = pydash.human_case(code)
+    return status_code, pydash.default_to(detail, "Unknown error")
```

Comment:
IMO this code 
```
detail = pydash.get(e.response, "Error.Message")
```
is easier to read than
```
try:
	detail = e.response["Error"]["Message"]
except KeyError:
	detail = None
```

Using `get` does not work because you can not use `Error.Message` as key name for the standard `dictionary.get(keyname, value)`.

That's why i prefer the pydash solution.